app/api/api_v1/endpoints/assignment_router.py

```diff
@@ -28,5 +37,32 @@ async def get_assignments(
         student_assignment = await StudentAssignmentService(db, student, assignment).get_student_assignment_schema()
         student_assignments.append(student_assignment)
         
+    return student_assignments
+
+@router.patch("/assignment/{assignment_name}", response_model=AssignmentSchema)
+def update_assignment_fields(
+    *,
+    db: Session = Depends(get_db),
+    assignment_name: str,
+    assignment_body: UpdateAssignmentBody
+):
+    # Assumption is that the Name is unique
+    assignment = db.query(AssignmentModel).filter_by(name=assignment_name).first()
+    if assignment is None:
+        raise HTTPException(status_code=404, detail="Assignment does not exist")
+
+    # Differentiate between fields that are set as None and fields that are not set at all
+    updated_set_fields = assignment_body.dict(exclude_unset=True)
+
+    if updated_set_fields.new_name is not None:
+        assignment.name = updated_set_fields.new_name
+    if updated_set_fields.directory_path is not None:
+        assignment.directory_path = updated_set_fields.directory_path
+
+    assignment.available_date = updated_set_fields.available_date
+    assignment.due_date = updated_set_fields.due_date
```

Comment:
Why are the first two vars behind an if check and the second two not?

Reply:
the first two vars are mandatory and non nullable fields in our model. so I am checking here before I update the value in the DB.

the second two vars are allowed to be None in our model. so if the prof specifically wants to set them as None, and provide that as part of the request body, we allow that